src/backends/llvm.ts

```diff
@@ -83,42 +133,75 @@ function emit_lookup(emitter: LLVMEmitter, emit_extern: (name: string, type: Typ
   }
 }
 
+function emit_extern(name: string, type: Type): llvm.Value {
+  if (_is_fun_type(type)) {
+    // The extern is a function. Wrap it in the clothing of our closure
+    // format (with no environment).
+    // TODO
+    throw "not implemented"
+  } else {
+    // An ordinary value. Just look it up by name.
+    // TODO
+    throw "not implemented"
+  }
+}
+
+// mostly a copy of emitter function
+function emit(emitter: LLVMEmitter, tree: ast.SyntaxNode): llvm.Value {
+  return emitter.emit_expr(tree, emitter);
+}
+
 ///////////////////////////////////////////////////////////////////
 // End Emit Functions & Redundant Funcs
 ///////////////////////////////////////////////////////////////////
 
+/**
+ * Create an alloca with the provided name in the entry block of the provided function
+ */
+function create_entry_block_alloca(func: llvm.Function, type: llvm.Type, name: string): llvm.Value {
+  // create builder and position after func's first instruction
+  let builder: llvm.Builder = llvm.Builder.create();
```

Comment:
Also, in js.ts it looked like you did "batch up" the variable declarations and do them all when entering a new function (i.e. inside the emit_fun function). 

I was originally going to do that, but since we don't really have functions yet, I just went with my current approach for now. I can try switching it over though once we start adding support for functions.